opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -54,6 +59,48 @@ def collect(self) -> Optional[_PointVarT]:
         pass
 
 
+class _AggregationFactory(ABC):
+    @abstractmethod
+    def _create_aggregation(self, instrument: Instrument) -> _Aggregation:
```

Comment:
Below we also return None when instrument matches nothing which is probably never going happen. 
```suggestion
    def _create_aggregation(self, instrument: Instrument) -> Optional[_Aggregation]:
```

Reply:
`_AggregationFactory` is being used for all the other regular `Aggregation`s as well, but they always return an `_Aggregation`. What should we do in this case for typing?

Reply:
I'll remove that return `None` and add a `raise Exception` instead. I only added that specific `return None` to please pylint.

Reply:
> `_AggregationFactory` is being used for all the other regular `Aggregation`s as well, but they always return an `_Aggregation`. What should we do in this case for typing?

Sorry, I don't see where the issue is here, can you provide more context, please?

Reply:
@ocelotl 

No issue here. Just responding to @srikanthccv 's suggestion. Your solution for raising an exception is fine.

Reply:
`_Aggregation` is fine if we are going to raise exception when there is no match.

Reply:
Added exception